### PR TITLE
fix: every it should has self ctx

### PR DESCRIPTION
**Pull request by killagu**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: lib/inject_context.js [hunk 1]
```diff
@@ -43,6 +44,54 @@ function injectContext(mocha) {
...
+              return resolve();
+            });
+          });
+        });
+      } catch (errSuite) {
+        return done(errSuite);
+      }
+      return next(i + 1);
+    }
+    next(0).catch(e => {
```
Comment (fengmk2) [inline lib/inject_context.js:88]: 仿佛看到早期的 koa 代码。

File: lib/inject_context.js [hunk 2]
```diff
@@ -43,6 +44,54 @@ function injectContext(mocha) {
...
+      suite.tests = tests;
+      return fn(errSuite);
+    }
+
+    async function next(i) {
+      const test = tests[i];
+      if (!test) {
+        return done();
+      }
+      suite.tests = [ test ];
```
Comment (fengmk2) [inline lib/inject_context.js:68]: suite.tests 里面的都是并行执行的对吧，修改之前？

Comment (killagu) [reply lib/inject_context.js:68]: 也是通过 next 这样的方式来顺序执行的。

File: lib/inject_context.js (hunk 1 above)
Comment (killagu) [reply lib/inject_context.js:88]: 差一点就要 `require('async')` 了。

File: lib/inject_context.js (hunk 2 above)
Comment (killagu) [reply lib/inject_context.js:68]: https://github.com/mochajs/mocha/blob/master/lib/runner.js#L709 这里。

**Pull request merged**